oak/common/policy.cc

```diff
@@ -41,12 +41,12 @@ oak::policy::Label DeserializePolicy(const std::string& policy_bytes) {
   return policy_proto;
 }
 
-oak::policy::Label AuthorizationBearerTokenPolicy(const std::string& authorization_token) {
+oak::policy::Label AuthorizationBearerTokenPolicy(const std::string& authorization_token_hmac) {
   oak::policy::Label label;
   auto* secrecy_tag = label.add_secrecy_tags();
   auto* integrity_tag = label.add_integrity_tags();
-  secrecy_tag->mutable_grpc_tag()->set_authorization_bearer_token(authorization_token);
-  integrity_tag->mutable_grpc_tag()->set_authorization_bearer_token(authorization_token);
+  secrecy_tag->mutable_grpc_tag()->set_authorization_bearer_token_hmac(authorization_token_hmac);
+  integrity_tag->mutable_grpc_tag()->set_authorization_bearer_token_hmac(authorization_token_hmac);
```

Comment:
Using the same label for integrity and secrecy seems weird. In many cases, such as the chat room, it seems that I would want to attach different secrecy and integrity labels (eg this message is from me, but can be read by anyone).

Also, integrity labels should not be forgeable. So at the very least the server should validate that the integrity label is aligned with the auth token. If the server needs to calculate the hmac of the auth token anyway, then their is no real need to pass the integrity label to the server in the first place.

Reply:
You are correct, integrity labels are currently meaningless, but so are secrecy labels, in that they don't actually control information flow (yet). I am postponing the actual implementation of that until we have more (all?) of the runtime in Rust, as I would not trust myself to get it right in C++.

Some discussion for this is in #420 , but probably we will need more granular issues once we get closer to an actual implementation.